src/server/game/AI/ScriptedAI/ScriptedCreature.cpp

```diff
@@ -200,6 +200,47 @@ void ScriptedAI::DoPlaySoundToSet(WorldObject* source, uint32 soundId)
     source->PlayDirectSound(soundId);
 }
 
+void ScriptedAI::AddThreat(Unit* victim, float amount, Unit* who)
+{
+    if (!who)
+        who = me;
+    who->AddThreat(victim, amount, nullptr, true, true);
```

Comment:
`AddThreat` is on `Unit` here because `AddThreat` was on `Unit` before. In the full PR, the method on `Unit` is merely [a forward to the threat manager](https://github.com/Treeston/TrinityCore/blob/364fe8af7618a9aad88da6e68618dc652513f032/src/server/game/Entities/Unit/Unit.cpp#L9360-L9364).